tethys_portal/views/api.py

```diff
@@ -1,100 +1,103 @@
-            return JsonResponse({})
-
-        return HttpResponse("Unauthorized", status=401)
-
+    user = request.user
     response_data = {
-        "username": request.user.username,
-        "firstName": request.user.first_name,
-        "lastName": request.user.last_name,
-        "email": request.user.email,
-        "isAuthenticated": request.user.is_authenticated,
-        "isStaff": request.user.is_staff,
+        "username": user.username,
+        "isAuthenticated": user.is_authenticated,
+        "isStaff": user.is_staff,
     }
-
-    # Generate gravatar URL if django_gravatar is available
-    try:
-        email = request.user.email if request.user.email else "tethys@example.com"
-        gravatar_url = get_gravatar_url(email, size=80)
-        response_data["gravatarUrl"] = gravatar_url
-    except Exception:
-        # If django_gravatar is not installed or fails, just skip adding gravatar URL
-        pass
-
-    return JsonResponse(response_data)
-
-
+    if not user.is_anonymous:
+        response_data["firstName"] = user.first_name
+        response_data["lastName"] = user.last_name
+        response_data["email"] = user.email
+        try:
+            email = user.email if user.email else "tethys@example.com"
+            gravatar_url = get_gravatar_url(email, size=80)
+            response_data["gravatarUrl"] = gravatar_url
+        except Exception:
+            pass
+    return Response(response_data)
+
+
+@api_view(["GET"])
+@permission_classes([AllowAny])
 def get_app(request, app):
     from tethys_apps.models import TethysApp
 
     package = app.replace("-", "_")
 
     try:
-        app = TethysApp.objects.get(package=package)
+        app_obj = TethysApp.objects.get(package=package)
     except TethysApp.DoesNotExist:
-        return JsonResponse({"error": f'Could not find app "{app}".'})
+        return Response({"error": f'Could not find app "{app}".'})
 
     metadata = {
-        "title": app.name,
-        "description": app.description,
-        "tags": app.tags,
-        "package": app.package,
-        "urlNamespace": app.url_namespace,
-        "color": app.color,
-        "icon": static(app.icon),
+        "title": app_obj.name,
+        "description": app_obj.description,
+        "tags": app_obj.tags,
+        "package": app_obj.package,
+        "urlNamespace": app_obj.url_namespace,
+        "color": app_obj.color,
+        "icon": static(app_obj.icon),
         "exitUrl": (
             reverse("app_library")
             if settings.MULTIPLE_APP_MODE
-            else reverse(app.index_url)
+            else reverse(app_obj.index_url)
         ),
-        "rootUrl": reverse(app.index_url),
-        "settingsUrl": f'{reverse("admin:index")}tethys_apps/tethysapp/{app.id}/change/',
+        "rootUrl": reverse(app_obj.index_url),
+        "settingsUrl": f'{reverse("admin:index")}tethys_apps/tethysapp/{app_obj.id}/change/',
     }
 
     if request.user.is_authenticated:
```

Comment:
Same question here about session and JWT. How is "is_authenticated" set without the session? Perhaps the JWT is optional for this endpoint?